uxarray/grid/coordinates.py

```diff
@@ -235,3 +234,69 @@ def _populate_lonlat_coord(grid):
             "long_name": "latitude of mesh nodes",
             "units": "degrees_north",
         })
+
+
+def _populate_centroid_coord(self):
+    """Finds the centroids using cartesian averaging of faces based off the
+    vertices."""
+
+    node_x = self.Mesh2_node_cart_x.values
+    node_y = self.Mesh2_node_cart_y.values
+    node_z = self.Mesh2_node_cart_z.values
+    face_nodes = self.Mesh2_face_nodes.values
+    nNodes_per_face = self.nNodes_per_face.values
+
+    mesh2_face_x, mesh2_face_y, mesh2_face_z = _construct_xyz_centroids(
+        node_x, node_y, node_z, face_nodes, nNodes_per_face)
```

Comment:
Doesn't that defeat the purpose of repopulate though? If they already have latlon and they want to repopulate latlon with our definition of centroid, this wouldn't do that. Same for if we do the same thing for the case when we have cartesian coordinates in our dataset.

Reply:
These are the following scenarios that we can end up in:
1. `Mesh2_face_x/y` don't exist
2. `Mesh2_face_x/y` exist, and we want to construct `Mesh2_face_cart_x/y/z` without repopulating the original `Mesh2_face_x/y`
3. `Mesh2_face_x/y` exist, and we want to re-compute them using this algorithm

#### Case 1
Nothing needs to be done here

#### Case 2
For this example, if the user wants the preserve the `Mesh2_face_x/y` that already exist within the `Grid` (which we should by default keep, since most of the time the centroid calculations that are parsed are sound), we should simply perform the Cartesian conversion to populate `Mesh2_face_cart_x/y/z` 

#### Case 2
I'm not 100% sure how we should handle this case, since given the way we have our attributes set up, there's no way to pass through the `repopulate=True` parameter. I think for now, we shouldn't worry about the repopulate tag (we can keep it until a good way comes up to expose it to the user)